bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,216 @@
+
+
+class TodoistClient(Client):
+    def __init__(self, token, filter):
+        self._api = TodoistAPI(token)
+        self.filter = filter
+
+    def get_projects(self):
+        all_projects = []
+        projects_iter = self._api.get_projects()
+        for projects in projects_iter:
+            for project in projects:
+                all_projects.append(project)
+        return all_projects
+
+    def get_sections(self):
+        all_sections = []
+        sections_iter = self._api.get_sections()
+        for sections in sections_iter:
+            for section in sections:
+                all_sections.append(section)
+        return all_sections
+
+    def get_users(self, project_id):
+        all_users = []
+        users_iter = self._api.get_collaborators(project_id)
+        for users in users_iter:
+            for user in users:
+                all_users.append(user)
+        return all_users
+
+    def get_issues(self):
+        tasks_iter = self._api.filter_tasks(query=self.filter)
+        for tasks in tasks_iter:
+            for task in tasks:
+                yield task
+
+
+class TodoistIssue(Issue):
+    ASSIGNEE = "todoistassignee"
+    ASSIGNER = "todoistassigner"
+    CONTENT = "todoistcontent"
+    DESCRIPTION = "todoistdescription"
+    DURATION = "todoistduration"
+    ID = "todoistid"
+    SECTION = "todoistsection"
+    URL = "todoisturl"
+
+    PRIORITY_MAP = {
+        4: "H",
+        3: "M",
+        2: "L",
+        1: None,
+    }
+
+    UDAS = {
+        ID: {
+            "type": "string",
+            "label": "Todoist ID",
+        },
+        CONTENT: {
+            "type": "string",
+            "label": "Todoist Content",
+        },
+        DESCRIPTION: {
+            "type": "string",
+            "label": "Todoist Description",
+        },
+        DURATION: {
+            "type": "string",
+            "label": "Todoist Duration",
+        },
+        SECTION: {
+            "type": "string",
+            "label": "Todoist Section",
+        },
+        ASSIGNEE: {
+            "type": "string",
+            "label": "Todoist Assignee",
+        },
+        ASSIGNER: {
+            "type": "string",
+            "label": "Todoist Assigner",
+        },
+        URL: {
+            "type": "string",
+            "label": "Todoist URL",
+        },
+    }
+
+    UNIQUE_KEY = (ID,)
+
+    # replace characters that cause escaping issues like [] and "
+    # this is a workaround for https://github.com/ralphbean/taskw/issues/172
+    def _unescape_content(self, content):
+        return (
+            content.replace('"', "'")  # prevent &dquote; in task details
+            .replace("[", self.config.char_open_bracket)  # prevent &open; and &close;
+            .replace("]", self.config.char_close_bracket)
+        )
+
+    def get_priority(self):
+        return self.PRIORITY_MAP.get(
+            self.record.priority,
+            self.config.default_priority
+        )
+
```

Comment:
This is practically identical to the inherited `get_priority` method inherited from `Issue` -- sorry that was unclear.
```suggestion
```

Reply:
The `self.record.get('priority')` from the inherited method fails because `record` is a `Task` and priority needs to be accessed as `self.record.priority`.

Reply:
Oh wow, I had missed that. I think it ought to be converted to a dict for consistency with other services.

Reply:
`record` is now a dict